Modix.Services.Test/DuplicitMessage/DuplicitMessageTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+using Modix.Services.Core;
+using Modix.Services.DuplicitMessage;
+using Modix.Services.Moderation;
+using Modix.Services.Test.Moderation;
+using Moq;
+using Moq.AutoMock;
+using NUnit.Framework;
+using Shouldly;
+
+namespace Modix.Services.Test.DuplicitMessage
+{
+    [TestFixture]
+    public class DuplicitMessageTests
+    {
+        private static AutoMocker CreateMocker()
+        {
+            var autoMocker = new AutoMocker();
+
+            var mockSelfUser = autoMocker.GetMock<ISocketSelfUser>();
+            mockSelfUser
+                .Setup(x => x.Id)
+                .Returns(1);
+
+            autoMocker.GetMock<ISelfUserProvider>()
+                      .Setup(x => x.GetSelfUserAsync(It.IsAny<CancellationToken>()))
+                      .Returns(Task.FromResult(mockSelfUser.Object));
+
+            return autoMocker;
+        }
+
+        private static ISocketMessage BuildTestMessage(AutoMocker autoMocker, string content, DateTimeOffset timestamp)
+        {
+            var mockMessage = autoMocker.GetMock<ISocketMessage>();
+
+            var mockGuild = autoMocker.GetMock<IGuild>();
+
+            var mockAuthor = autoMocker.GetMock<IGuildUser>();
+            mockAuthor
+                .Setup(x => x.Guild)
+                .Returns(mockGuild.Object);
+            mockAuthor
+                .Setup(x => x.Id)
+                .Returns(2);
+
+            mockMessage
+                .Setup(x => x.Author)
+                .Returns(mockAuthor.Object);
+
+            var mockChannel = autoMocker.GetMock<IMessageChannel>();
+            mockChannel
+                .As<IGuildChannel>()
+                .Setup(x => x.Guild)
+                .Returns(mockGuild.Object);
+            mockMessage
+                .Setup(x => x.Channel)
```

Comment:
Why mock individual fields instead of just creating a message instance?

Reply:
I would need to create new class implementing `ISocketMessage` or just make `SocketMessageAbstraction` `public` to use it, but this is way it is faster